classes/ErrorHandler.php

```diff
@@ -0,0 +1,164 @@
+    {
+        error_reporting(E_ALL);
+        set_error_handler(array($this, 'errorHandler'));
+        set_exception_handler(array($this, 'exceptionHandler'));
+        register_shutdown_function(array($this, "fatalHandler"));
+    }
+
+    /**
+     * Function retrieving uncaught exceptions
+     * 
+     * @param Throwable $e
+     */
+    public function exceptionHandler($e)
+    {
+        $message = get_class($e) .': '. $e->getMessage();
+        $this->report($e->getFile(), $e->getLine(), Logger::CRITICAL, $message, $e->getTraceAsString(), true);
+    }
+
+    /**
+     * Function called by PHP errors, forwarding content to the ajax response
+     *
+     * @param int $errno
+     * @param string $errstr
+     * @param string $errfile
+     * @param string $errline
+     * @return boolean
+     */
+    public function errorHandler($errno, $errstr, $errfile, $errline)
+    {
+        if (!(error_reporting() & $errno)) {
+            // This error code is not included in error_reporting, so let it fall
+            // through to the standard PHP error handler
+            return false;
+        }
+
+        switch ($errno) {
+            case E_USER_ERROR:
+                return false; // Will be taken by fatalHandler
+            case E_USER_WARNING:
+            case E_WARNING:
+                $type = Logger::WARNING;
+                break;
+            case E_NOTICE:
+            case E_USER_NOTICE:
+                $type = Logger::NOTICE;
+                break;
+            default:
+                $type = Logger::DEBUG;
+                break;
+        }
+
+        $this->report($errfile, $errline, $type, $errstr);
+        return true;
+    }
+
+    /**
+     * Fatal error from PHP are not taken by the error_handler. We must check if an error occured
+     * during the script shutdown.
+     */
+    public function fatalHandler()
+    {
+        $lastError = error_get_last();
+        $trace = isset($lastError['backtrace']) ? var_export($lastError['backtrace'], true) : null;
+        if ($lastError && in_array($lastError['type'], array(E_ERROR, E_PARSE, E_CORE_ERROR, E_COMPILE_ERROR, E_USER_ERROR), true)) {
+            $this->report($lastError['file'], $lastError['line'], Logger::CRITICAL, $lastError['message'], $trace, true);
+        }
+    }
+
+    /**
+     * Create a json encoded
+     * @param string $log
+     * @return string
+     */
+    public function generateJsonLog($log)
+    {
+        return '{"nextErrors":["'.addcslashes($log, '"\\').'"],"error":true,"next":"error"}';
+    }
+
+    /**
+     * Forwards message to the main class of the upgrade
+     * 
+     * @param string $file
+     * @param int $line
+     * @param int $type Level of criticity
+     * @param string $message
+     * @param bool $display
+     */
+    protected function report($file, $line, $type, $message, $trace = null, $display = false)
+    {
+        if ($type >= Logger::CRITICAL) {
+            http_response_code(500);
```

Comment:
Can use Symfony const  https://api.symfony.com/3.4/Symfony/Component/HttpFoundation/Response.html instead of 500

Reply:
I can't, this module can be used un PrestaShop 1.6, which does not provide Symfony